h2o-automl/src/main/java/ai/h2o/automl/leaderboard/Leaderboard.java

```diff
@@ -198,42 +213,39 @@ public Frame leaderboardFrame() {
   }
 
   /** Return the number of models in this Leaderboard. */
-  int getModelCount() { return getModelKeys().length; }
+  public int getModelCount() { return getModelKeys() == null ? 0 : getModelKeys().length; }
 
   /**
    * @return list of models sorted by the default metric for the model category
    */
   public Model[] getModels() {
-    Key<Model>[] modelKeys = getModelKeys();
-    if (modelKeys == null || 0 == modelKeys.length) return new Model[0];
-    return getModelsFromKeys(modelKeys);
+    if (getModelCount() == 0) return new Model[0];
+    return getModelsFromKeys(getModelKeys());
   }
 
   /**
    * @return list of models sorted by the given metric
    */
   public Model[] getModelsSortedByMetric(String metric) {
-    Key<Model>[] modelKeys = sortModels(metric);
-    if (modelKeys == null || 0 == modelKeys.length) return new Model[0];
-    return getModelsFromKeys(modelKeys);
+    if (getModelCount() == 0) return new Model[0];
+    return getModelsFromKeys(sortModels(metric));
```

Comment:
@seb-h2o wdyt about changing `sortModels` to `getSortedModelKeys`? or at least `sortModelKeys`?

Reply:
- `sortModels` is mainly delegating to `ModelMetrics.sortModelsByMetric`, so it's consistent internally at least.
- I tend to avoid `get___` for methods that do something instead of just returning an object stored somewhere. Otherwise, any method that doesn't return `void` would be potentially named `getSomethingAfterDoingSomethingElse`.
- `sortModels` is private, so I'm less picky about its naming.
- I actually prefer `sortModels` to `sortModelKeys` because that's really the models we're sorting somehow (we need the model to obtain its metric and so on...), if we were just sorting keys we would have choice between lexicographic order, and... it's quite limited.

Reply:
@seb-h2o 
> I tend to avoid get___ for methods that do something instead of just returning an object stored somewhere.

does mention activity stays visible to the outside world? if no then I would say it should not matter whether we just retrieve from storage or compute something to prepare result. 

> I actually prefer sortModels to sortModelKeys because that's really the models we're sorting somehow (we need the model to obtain its metric and so on.

i think it is result(returning values) that defines naming.... we should not care **how** it is done

> sortModels is private, so I'm less picky about its naming

agree, but to care about developers as well we better be equally picky :)

Reply:
> we should not care how it is done
I agree with that, but that's not what I mean. I mean we are really sorting the models, not its keys. It's just that the keys are just what you need to access the models then...
> result defines naming
so many examples where this is not true, it's mainly true for accessors. But definitely not a requirement imo, especially with a strongly types language.